openshift/templates/tools/pipeline.yaml

```diff
@@ -0,0 +1,69 @@
+kind: "Template"
+apiVersion: "v1"
+metadata:
+  name: "${NAME}-pipeline-template"
+  creationTimestamp: null
+objects:
+  - kind: "BuildConfig"
+    apiVersion: "v1"
+    metadata:
+      name: "${NAME}-pipeline"
+      creationTimestamp: null
+      labels:
+        app: "${NAME}-pipeline"
+        name: "${NAME}-pipeline"
+        template: "${NAME}-pipeline"
+    spec:
+      triggers:
+        - type: "GitHub"
+          github:
+            secret: "${GITHUB_WEBHOOK_SECRET}"
+        - type: "Generic"
+          generic:
+            secret: "${GENERIC_WEBHOOK_SECRET}"
+      runPolicy: "Parallel"
+      source:
+        type: "Git"
+        git:
+          uri: ${GIT_REPO_URL}
+          ref: ${GIT_REF}
+        contextDir: ${SOURCE_CONTEXT_DIR}
+      strategy:
+        type: "JenkinsPipeline"
+        jenkinsPipelineStrategy:
+          jenkinsfilePath: "${JENKINSFILE_PATH}"
+      output: {}
+      resources: {}
+      postCommit: {}
+    status:
+      lastVersion: 0
+parameters:
+  - name: "NAME"
```

Comment:
How are these params without names being filled in at the moment? I'm using .param files committed to source, is there another method that I should be aware of?

Reply:
I suspect the CLI.